CodeEdit/Features/LSP/Service/LSPService.swift

```diff
@@ -0,0 +1,286 @@
+/// handling them as they occur. The `LSPService` class also provides functionality to start
+/// and stop individual language servers, as well as to stop all running servers.
+///
+/// ## Example Usage
+///
+/// ```swift
+/// @Service var lspService
+///
+/// try await lspService.startServer(
+///    for: .python,
+///    projectURL: projectURL,
+///    workspaceFolders: workspaceFolders
+/// )
+/// try await lspService.stopServer(for: .python)
+/// ```
+///
+/// ## Completion Example
+///
+/// ```swift
+/// func testCompletion() async throws {
+///     do {
+///         guard var languageClient = self.languageClient(for: .python) else {
+///             print("Failed to get client")
+///             throw ServerManagerError.languageClientNotFound
+///         }
+///
+///         let testFilePathStr = ""
+///         let testFileURL = URL(fileURLWithPath: testFilePathStr)
+///
+///         // Tell server we opened a document
+///         _ = await languageClient.addDocument(testFileURL)
+///
+///         // Completion example
+///         let textPosition = Position(line: 32, character: 18)  // Lines and characters start at 0
+///         let completions = try await languageClient.requestCompletion(
+///             document: testFileURL.absoluteString,
+///             position: textPosition
+///         )
+///         switch completions {
+///         case .optionA(let completionItems):
+///             // Handle the case where completions is an array of CompletionItem
+///             print("\n*******\nCompletion Items:\n*******\n")
+///             for item in completionItems {
+///                 let textEdits = LSPCompletionItemsUtil.getCompletionItemEdits(
+///                     startPosition: textPosition,
+///                     item: item
+///                 )
+///                 for edit in textEdits {
+///                     print(edit)
+///                 }
+///             }
+///
+///         case .optionB(let completionList):
+///             // Handle the case where completions is a CompletionList
+///             print("\n*******\nCompletion Items:\n*******\n")
+///             for item in completionList.items {
+///                 let textEdits = LSPCompletionItemsUtil.getCompletionItemEdits(
+///                     startPosition: textPosition,
+///                     item: item
+///                 )
+///                 for edit in textEdits {
+///                     print(edit)
+///                 }
+///             }
+///
+///             print(completionList.items[0])
+///
+///         case .none:
+///             print("No completions found")
+///         }
+///
+///         // Close the document
+///         _ = await languageClient.closeDocument(testFilePathStr)
+///     } catch {
+///         print(error)
+///     }
+/// }
+/// ```
+@MainActor
+final class LSPService: ObservableObject {
+    internal let logger: Logger = Logger(subsystem: Bundle.main.bundleIdentifier ?? "", category: "LSPService")
```

Comment:
`internal` keyword here in the CodeEdit repo is redundant, I think

Reply:
Removed